source/common/network/tcp_keepalive_option_impl.cc

```diff
@@ -0,0 +1,73 @@
+#include "common/network/tcp_keepalive_option_impl.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/network/address_impl.h"
+
+namespace Envoy {
+namespace Network {
+bool TcpKeepaliveOptionImpl::setOption(Network::Socket& socket,
+                                       Network::Socket::SocketState state) const {
+  if (state == Socket::SocketState::PreBind) {
```

Comment:
Will add a new state, perhaps PreConnect, since PreBind is a bit too vague for this option.